### ISSUE TITLE
[feature/composer]: Issue right after initial git checkout and ddev start - Help wanted

### ISSUE THREAD

**Issue by stevleibelt** [s9y/Serendipity#833]
Hey,

my journey started with [PR832](https://github.com/s9y/Serendipity/pull/832) containing failed PHP Tests.

I wanted to quickly have a look which turned out to yet my current Problem.

What I did was:

```bash
git clone https://github.com/s9y/Serendipity
cd Serendipity
git checkout dependabot/composer/feature/composer/smarty/smarty-5.0.0
ddev start
ddev composer install
```

Than I've opened my browser pointing to `http://serendipity.ddev.site`.   
Once opened this location, I got redirected to `http://serendipity.ddev.site//serendipity_admin.php`.

This page shows me the following error:

> Fatal error: Uncaught Error: Undefined constant "LANG_CHARSET" in /var/www/html/public/serendipity_admin.php:11 Stack trace: #0 {main} thrown in /var/www/html/public/serendipity_admin.php on line 11

Because of that, I've started digging. But before sharing my results, I just want know if I've missed an important step? I tried to understand the [branch readme](https://github.com/s9y/Serendipity/blob/feature/composer/README.markdown#concept) file but could not find the missing todo.

What I've figured out on my machine (just for the fun, I am using arch btw. ;-)):

* `public/serendipity_config.inc.php` war parsed until line [278](https://github.com/s9y/Serendipity/blob/feature/composer/public/serendipity_config.inc.php#L278)   
* Because of that, line [447](https://github.com/s9y/Serendipity/blob/feature/composer/public/serendipity_config.inc.php#L447) can not be reached
* If I add `defined('LANG_CHARSET') or @define('LANG_CHARSET', 'ISO-8859-1');` bevor line [278](https://github.com/s9y/Serendipity/blob/feature/composer/public/serendipity_config.inc.php#L278), I ran into another issue I could not solve
* `lang.inc.php` has the line [48](https://github.com/s9y/Serendipity/blob/feature/composer/src/include/lang.inc.php#L48) where `$charset` is `""` in my project
* If I replace `S9Y_INCLUDE_PATH . 'lang/'` with `S9Y_INCLUDE_PATH . '../lang/'`, I run into smarty include errors
* This was the moment where I've realized that I must do something perfectly wrong

Would you please be so kind and put my nose into the right direction?
Since there is this ddev support, it would be a shame to not try to support you.

Best regards,
Stev

**Comment by fe-hicking** [comment]
Sooo, a couple of things to unravel:

feature/composer is the main branch where I tried initial ground work for proper composer integration (s9y as a composer package).

That has a nice concept, see its DREAMME, uhm, README. But it doesn't work yet.

The ddev integration should be solid, but the core still needs a lot of include changes, because I had to slaughter the whole directory structure, if a composer package integration is to be achieved.

The branch you checked out is a dependabot automatic integration of composer updates, that creates automatic branches. It recently updated to smarty 5.0. That will/would fail horrificly at this point.

So, if you want to "just" check out the "working" version, use the "main" branch.

If you're in for some fun with me and a lot of unpaid work hours, check out the feature/composer branch and the readme, and lets see how we can divide and concquer.
